frontend/workflows/projectSelector/src/project-selector.tsx

```diff
@@ -295,48 +300,83 @@ const ProjectSelector = ({ onError }: ProjectSelectorProps) => {
     setCustomProject(event.target.value);
   };
 
+  // Autorefresh will eventually be passed via hooks to the cards to enforce their update intervals
+  const [autoRefresh, setAutoRefresh] = React.useState<boolean>(true);
+
+  // useEffect that handles auto-refreshing / reloading.
+  React.useEffect(() => {
+    // This way of autorefreshing is also used in other places.
+    if (autoRefresh) {
+      // hardcode to 30 seconds
+      const interval = setInterval(() => hydrateProjects(state, dispatch, fromShortLink), 30000);
+      return () => clearInterval(interval);
+    }
+    return () => {};
+  }, [hasError, state, autoRefresh]);
```

Comment:
for my understanding - why is `hasError` included?

Reply:
oh yeah actually I realized that its just coming from the `state`, so we don't actually need it. good catch, thanks.